sha1/src/checked/ubc_check.rs

```diff
@@ -0,0 +1,1179 @@
+//! Direct translation of the C code
+//! Original license and comments:
+//!
+//! Copyright 2017 Marc Stevens <marc@marc-stevens.nl>, Dan Shumow <danshu@microsoft.com>
+//! Distributed under the MIT Software License.
+//! See accompanying file LICENSE.txt or copy at
+//! https://opensource.org/licenses/MIT
+//!
+//! this file was generated by the 'parse_bitrel' program in the tools section
+//! using the data files from directory 'tools/data/3565'
+//!
+//! sha1_dvs contains a list of SHA-1 Disturbance Vectors (DV) to check
+//! dvType, dvK and dvB define the DV: I(K,B) or II(K,B) (see the paper)
+//! dm[80] is the expanded message block XOR-difference defined by the DV
+//! testt is the step to do the recompression from for collision detection
+//! maski and maskb define the bit to check for each DV in the dvmask returned by ubc_check
+//!
+//! ubc_check takes as input an expanded message block and verifies the unavoidable bitconditions for all listed DVs
+//! it returns a dvmask where each bit belonging to a DV is set if all unavoidable bitconditions for that DV have been met
+//! thus one needs to do the recompression check for each DV that has its bit set
```

Comment:
Similar to above, this should probably be moved to be a doc comment on `fn ubc_check`

Reply:
fixed